Nodejs/Tests/TestAdapter.Tests/TestProjectFactory.cs

```diff
@@ -0,0 +1,129 @@
+        private string projectNameString => Enum.GetName(typeof(ProjectName), this.projectName);
+
+        public TestProjectFactory(ProjectName projectName)
+        {
+            this.projectName = projectName;
+        }
+
+        public string GetProjectDirPath()
+        {
+            return Path.Combine(Directory.GetCurrentDirectory(), $@"..\..\..\..\MockProjects\{projectNameString}");
+        }
+
+        public string GetProjectFilePath()
+        {
+            switch (this.projectName)
+            {
+                case ProjectName.NodeAppWithTestsConfiguredOnProject:
+                case ProjectName.NodeAppWithTestsConfiguredPerFile:
+                case ProjectName.NodeAppWithAngularTests:
+                    return Path.Combine(this.GetProjectDirPath(), $"{projectNameString}.njsproj");
+            }
+
+            throw new ArgumentException();
+        }
+
+        public List<TestCase> GetTestCases()
+        {
+            switch (this.projectName)
+            {
+                case ProjectName.NodeAppWithTestsConfiguredOnProject:
+                    return new List<TestCase>(this.GetTestCases(SupportedFramework.Mocha));
+
+                case ProjectName.NodeAppWithTestsConfiguredPerFile:
+                    var result = new List<TestCase>();
+                    result.AddRange(this.GetTestCases(SupportedFramework.Jasmine));
+                    result.AddRange(this.GetTestCases(SupportedFramework.ExportRunner));
+                    result.AddRange(this.GetTestCases(SupportedFramework.Jest));
+                    result.AddRange(this.GetTestCases(SupportedFramework.Mocha));
+                    result.AddRange(this.GetTestCases(SupportedFramework.Tape));
+
+                    return result;
+
+                case ProjectName.NodeAppWithAngularTests:
+                    return new List<TestCase>(this.GetTestCases(SupportedFramework.Angular));
+
+            };
+
+            throw new InvalidOperationException();
+        }
+
+        private List<TestCase> GetTestCases(SupportedFramework testFramework)
+        {
+            var configPath = this.projectName == ProjectName.NodeAppWithAngularTests
+                ? this.GetProjectDirPath()
+                : null;
+
+            var testCaseFactory = new TestCaseFactory(
+                testFramework,
+                this.GetProjectDirPath(),
+                this.GetProjectFilePath(),
+                configPath);
+
+            switch (testFramework)
+            {
+                case SupportedFramework.Jasmine:
+                    return new List<TestCase>()
+                    {
+                        testCaseFactory.GetTestCase("JasmineUnitTest.js::Test Suite 1::Test 1", "Test 1", 1, Path.Combine(this.GetProjectDirPath(), "JasmineUnitTest.js")),
+                        testCaseFactory.GetTestCase("JasmineUnitTest.js::Test Suite 1::Test 2", "Test 2", 1, Path.Combine(this.GetProjectDirPath(), "JasmineUnitTest.js")),
```

Comment:
```suggestion
                        var filePath = Path.Combine(this.GetProjectDirPath(), "JasmineUnitTest.js");
                        testCaseFactory.GetTestCase("JasmineUnitTest.js::Test Suite 1::Test 1", "Test 1", 1, filePath),
                        testCaseFactory.GetTestCase("JasmineUnitTest.js::Test Suite 1::Test 2", "Test 2", 1, filePath)),
```

Reply:
Added suggestion. I had to add a local scope to the switch tho.